pkgs/native_assets_cli/lib/src/api/build_output.dart

```diff
@@ -13,30 +13,99 @@ import '../model/metadata.dart' as model;
 import 'asset.dart';
 import 'dependencies.dart';
 import 'metadata.dart';
+import 'target.dart';
 
+/// The output of a `build.dart` invocation.
+///
+/// The Dart and Flutter SDK consume this data.
 abstract class BuildOutput {
-  /// Time the build this output belongs to started.
+  /// The [timestamp] indicates the time the build this output belongs to
+  /// started.
   ///
-  /// Rounded down to whole seconds, because [File.lastModified] is rounded
-  /// to whole seconds and caching logic compares these timestamps.
+  /// The [timestamp] is rounded down to whole seconds, because
+  /// [File.lastModified] is rounded to whole seconds and caching logic compares
+  /// these timestamps.
   DateTime get timestamp;
+
+  @Deprecated('Use assets2')
   List<Asset> get assets;
```

Comment:
It's complicated.

Flutter tools and dartdev both depend on the implementations in model. Any breaking change there is very hard due to https://github.com/dart-lang/native/blob/main/pkgs/native_assets_builder/CONTRIBUTING.md#rolling .

The reason that both `build.dart` scripts and Flutter tools and dartdev use the same underlying classes was to make it easier to test the serialization/deserialization for `BuildConfig` and `BuildOutput`. Maybe this was a mistake. If we'd completely separate the code being used in `build.dart` files and in the Dart/Flutter tooling, then we could _simply_ make a breaking change for `build.dart` script imports. However, it would become harder to test that (de)serialization in the protocol is correct.

I'm open to suggestions, because the current approach is rather suboptimal.

(Side note, one simplification would be to fold native_assets_builder completely into package:native_assets_cli/native_assets_cli_internal. But that means users on `pub get` would be pulling all kinds of code in they'd never use.)